azurelinuxagent/common/osutil/default.py

```diff
@@ -908,7 +907,10 @@ def is_missing_default_route(self):
         return True
 
     def get_if_name(self):
-        return self.get_first_if()[0]
+        ifname = ''
+        while len(ifname) < 2:
```

Comment:
Why? Looks to me like all exit paths from `self.get_first_if()` return a 2-tuple.

Reply:
Yes, but the callers for `get_if_name` do not all deal with `''` very well. Or did I misunderstand your concern?

Reply:
I misread how ifname was being set, so my original comment is withdrawn.

I have a new comment, though: if there's no interface other than loopback, this code will run in a very tight loop. Clearly that scenario occurs, though, since there was strong complaint about throwing an exception in this case; were this a never-happens scenario, no one would have complained.

Reply:
This is true, and matches with the existing behavior in `get_mac_addr` which we hit occasionally too. The loop is somewhat controlled by reaching out to disk but if you prefer we could add a sleep. What would you suggest?

Reply:
I'm just surprised this isn't a problem in practice. If you tell me it works out okay, I'm fine with that. Otherwise, I'd suggest exponential backoff and logging. (No point in trying to emit telemetry, since the essence of the problem is "I ain't got no network".) Hammering on the cpu in a tight loop might actually slow down whatever in-guest process will eventually provide a NIC.

Reply:
I am going to merge for the moment, and keep an eye on the bvts for anything unexpected. We can always add the backoff if necessary.